app/main.py

```diff
@@ -1,21 +1,25 @@
+from copy import deepcopy
 from typing import Any, Hashable
 
 
 class Dictionary:
     def __init__(self) -> None:
         self.size = 8
+        self.lenght = 0
         self.dict_ = [None for i in range(self.size)]
 
     def resize(self) -> None:
-        temp_ = [el for el in self.dict_ if el is not None]
-        if len(temp_) > (self.size * (2 / 3)):
+        if len(self) > (self.size * (2 / 3)):
+            self.lenght = 0
+            temp_ = deepcopy(self.dict_)
```

Comment:
Don't use deepcopy here. You probably don't want the objects in your dict to suddenly get replaced with copies when it gets resized.